book/src/building/swift-package/README.md

```diff
@@ -0,0 +1,246 @@
+// src/lib.rs
+#[swift_bridge::bridge]
+mod ffi {
+    extern "Rust" {
+        fn hello_rust();
+    }
+}
+
+fn hello_rust() -> String {
+    String::from("Hello Rust!")
+}
+```
+
+Add a new `build.rs` file:
+```rust
+// build.rs
+
+use std::path::PathBuf;
+
+fn main() {
+    let out_dir = PathBuf::from("./generated");
+
+    let bridges = vec!["src/lib.rs"];
+    for path in &bridges {
+        println!("cargo:rerun-if-changed={}", path);
+    }
+
+    swift_bridge_build::parse_bridges(bridges)
+        .write_all_concatenated(out_dir, env!("CARGO_PKG_NAME"));
+}
+```
+
+Build the project for the desired platforms:
+
+```bash
+cargo build --target x86_64-apple-darwin
+cargo build --target aarch64-apple-ios
+cargo build --target x86_64-apple-ios
+```
+
+## Creating the XCFramework
+
+Go back to the root of the project and make a new directory.
+
+```bash
+mkdir MyFramework && cd $_
+```
+
+Copy the generated libraries and the headers to this folder:
+
+```
+MyFramework
+├── include
+│   ├── SwiftBridgeCore.h
+│   ├── module.modulemap
+│   └── my_rust_lib.h
+├── ios
+│   └── libmy_rust_lib.a
+├── macos
+│   └── libmy_rust_lib.a
+└── simulator
+    └── libmy_rust_lib.a
+```
+
+Add `include/module.modulemap`:
+
+```modulemap
+module MyRustLib {
+    header "my_rust_lib.h"
+    header "SwiftBridgeCore.h"
+    export *
+}
+```
+
+Now it is time to build the xcframework:
+
+```bash
+xcodebuild -create-xcframework \
+    -library simulator/librust.a \
+    -headers include \
+    -library ios/librust.a \
+    -headers include \
+    -library macos/librust.a \
+    -headers include \
+    -output MyRustLib.xcframework
+```
+
+*The order of the `library` tags is important*
+
+## Creating the Swift package
+
+Go back to the root of the project and create a new Swift package, copy the xcframework and generated swift files to it:
+
+```
+MySwiftPackage
+├── Sources
+│   └── MySwiftPackage
+│       └── SwiftBridgeCore.swift
+│       └── my_rust_lib.swift
+├── MyRustLib.b.xcframework
+└── Package.swift
+```
+
+Add the framework as a binary target
+```swift
+// Package.swift
+
+let package = Package(
+    name: "MySwiftPackage",
+    products: [
+        .library(
+            name: "MySwiftPackge",
+            targets: ["MySwiftPackge"]),
+    ],
+    dependencies: [
+
+    ],
+    targets: [
+        .binaryTarget(
+            name: "MyRustLib",
+            path: "MyRustLib.xcframework"
+        ),
+        .target(
+            name: "MySwiftPackge",
+            dependencies: ["MyRustLib"]),
+    ]
+)
+```
+
+We need to make a few adjustments to the `.swift` files:
+
+```swift
+// Sources/MySwiftPackage/SwiftBridgeCore.swift
+import MyRustLib
+
+// ...
+```
+
+```swift
+// Sources/MySwiftPackage/my_rust_lib.swift
+import MyRustLib
+```
+
+We can choose to either make the functions and classes in `SwiftBridgeCore.swift` public, or to convert the return types of our functions to types recognized by swift, so in this case, do either of the following: 
```

Comment:
Oh ok, so the issue here is that the `toString()` function and other functions are private?

You can make `toString` public here:

https://github.com/chinedufn/swift-bridge/blob/771fc5cae52b0206b43135ee3adbc286f6cd64bd/src/std_bridge/string.swift#L27-L34

Then I think we can get rid of this suggestion?